add documentation of getPtr to overview.md (#375)
NVIDIA/cuda-python: cuda_bindings/docs/source/overview.md

Screenshot of Nsight Compute CLI output of CUDA Python example.
```

> ## Getting the address of underlying C objects from the low-level bindings

Reviewer: This section is not used anywhere in this page. Can we move it to a new page, say "Q&A", "Tips & tricks", "Common pitfalls", etc, so that we can collect this and other pending doc updates in one easy-to-find place for binding users?

> Within the low-level object wrappers there are a number of cdef classes which effectively provide a PyObject interface to CUDA types. For example, the CUdevice type is defined by:

Reviewer: I am a bit hesitant to mention that it is a `cdef class` here. The fact that it's a Cython extension class should be an implementation detail. Can we just say the CUDA types are exposed to Python as Python classes, with certain established methods?

```cython
cdef class CUdevice:
    ... 

    def __int__(self):
        return <int>self._ptr[0]
    def getPtr(self):
        return <void_ptr>self._ptr
```

> There is an important distinction between the `getPtr()` method and the behaviour of `__int__()`. If the user wants to get the address of the underlying C object, wrapped in the cdef python class, they should call `int(CUdeviceInstance)`, which returns a pointer to the object, while calling `CUdeviceInstance.getPtr()` returns the `void**` address of the pointer to the object.

Reviewer: @vzhurba01 can correct me, I think we need `getPtr()` only when preparing for kernel arguments (https://github.com/NVIDIA/cuda-python/issues/127). Perhaps we should mention this method in that context?

Reviewer 2: Neither of the examples nor unit tests under test_kernelParams demonstrate to the user how to use `getPtr()` to prepare kernel arguments. It's true that the bindings leverage the `getPtr()` in a [utils.pyx](https://github.com/NVIDIA/cuda-python/blob/main/cuda_bindings/cuda/bindings/_lib/utils.pyx.in#L50), but how that's done is a cython implementation detail.

Right now we give examples of how to prep kernel arguments through the use of `ctypes`, `numpy` and the cuda.bindings assisted packing as demonstrated in [test_kernelParams](https://github.com/NVIDIA/cuda-python/blob/main/cuda_bindings/tests/test_kernelParams.py#L439). I don't think we should mention `getPtr()` in the context of preparing kernel arguments unless the three methods above are insufficient and we plan on adding examples/doc on how to use `getPtr()` to pack.

Reviewer: I see, it sounds reasonable to me. So you are saying we shouldn't even mention `getPtr`?

Reviewer 2: I think using Keenan's description for what `getPtr` does is sufficient.

Screenshot of Nsight Compute CLI output of CUDA Python example.
```

## Getting the address of underlying C objects from the low-level bindings

> Within the low-level object wrappers there are a number of cdef classes which effectively provide a PyObject interface to cuda types such as CUdevice, which will be used an example
> in this section. The definition is as follows:

Reviewer 2: ```suggestion
Within the low-level object wrappers there are a number of cdef classes which effectively provide a PyObject interface to CUDA types. For example, the CUdevice type is defined by:
```
Perhaps phrased this way?

Author: yeah I prefer what you wrote
